router/core/factoryresolver.go

```diff
@@ -69,8 +77,7 @@ func NewDefaultFactoryResolver(
 	log *zap.Logger,
 	enableSingleFlight bool,
 	enableNetPoll bool,
-	natsPubSubBySourceID map[string]pubsub_datasource.NatsPubSub,
-	kafkaPubSubBySourceID map[string]pubsub_datasource.KafkaPubSub,
+	instanceData InstanceData,
```

Comment:
you have passed instanceData to contstructor function, but haven't assigned it to DefaultFactoryResolver field